adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -215,20 +215,26 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
 
                         // check that MQTT client tries to publish on topic with device_id same as on connection
                         if (resource.getResourceId().equals(endpoint.clientIdentifier())) {
+                            // check credentials for valid authentication
+                            // TODO: how to determine the type? I would think: configurable for MQTT, hashed-password, preshared key maybe
```

Comment:
Oops, destroyed the new lines, should have been:
`hono:`
`   mqtt:`
`       authentication:`
`       - hashed-password`
`       - preshared-key`
and so on...